GAE/src/org/waterforpeople/mapping/app/web/SurveyRestServlet.java

```diff
@@ -393,6 +395,32 @@ private SurveyInstanceDto findSurveyInstance(Long surveyInstanceId) {
         return dtoList;
     }
 
+    /**
+     * lists all questions for a given survey
+     *
+     * @param surveyId
+     * @return
+     */
+    private List<QuestionDto> listSurveyQuestions(Long surveyId) {
+        List<Question> questions = qDao.listQuestionsBySurvey(surveyId); //useless ordering
+        List<QuestionDto> dtoList = new ArrayList<QuestionDto>();
+        if (questions != null) {
+            for (Question q : questions) {
+                QuestionDto dto = new QuestionDto();
+                DtoMarshaller.copyToDto(q, dto);
+                if (q.getType().equals(Question.Type.CASCADE) && q.getCascadeResourceId() != null) {
```

Comment:
Maybe we can take this part out of here as its not to do with listing questions.  Maybe create a separate function and pass it the list of questions and then do this.